### PR TITLE
[runtime] Ephemeral Allocator Misses to Allocate Last Port on Private Range

### PR DESCRIPTION
**Pull request by ppenna**
## Description

This PR closes https://github.com/microsoft/demikernel/issues/1039

## Summary of Changes

- Included last port in the allocator
- Changed shuffling to occur only on release mode
- Cleaned up dead code

### CODE REVIEW

**Review by anandbonde — APPROVED** [review]
Looks good.

File: src/rust/runtime/network/ephemeral.rs [hunk 1]
```diff
@@ -12,11 +17,10 @@ use ::rand::prelude::{
 // Constants
 //======================================================================================================================
 
+/// First private port.
```
Comment (anandbonde) [inline src/rust/runtime/network/ephemeral.rs:20]: It would be good to comment why these values are chosen for these consts.

Comment (ppenna) [reply src/rust/runtime/network/ephemeral.rs:20]: These come from the RFC itself, sure I can link the RFC here.

File: src/rust/runtime/network/ephemeral.rs [hunk 2]
```diff
@@ -96,21 +90,15 @@ impl EphemeralPorts {
...
-                SmallRng::seed_from_u64(EPHEMERAL_PORT_SEED)
-            }
-            #[cfg(not(debug_assertions))]
-            {
-                SmallRng::from_entropy()
-            }
-        };
         let mut ports: Vec<u16> = Vec::<u16>::new();
-        for port in FIRST_PRIVATE_PORT..LAST_PRIVATE_PORT {
+        for port in (FIRST_PRIVATE_PORT..=LAST_PRIVATE_PORT).rev() {
```
[the remark this answers is not in the document]
Comment (anandbonde) [reply src/rust/runtime/network/ephemeral.rs:94]: Sounds good.

**Pull request merged**